sycl/doc/extensions/proposed/sycl_ext_oneapi_group_load_store.asciidoc

```diff
@@ -391,10 +392,42 @@ For example, we can assert that `input` is a contiguous iterator:
 
 If `input` isn't a contiguous iterator, the behavior is undefined.
 
+==== Groups partitioning
+
+The following properties  is introduced to be used
+as a hint that implementation can use get_max_local_range():
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+struct full_work_group_hint {
+  using value_t =
+      property_value<full_work_group_hint>;
+};
+
+inline constexpr full_work_group_hint::value_t full_work_group;
+
+struct full_sub_group_hint {
+  using value_t =
+      property_value<full_sub_group_hint>;
+};
+
+inline constexpr full_sub_group_hint::value_t full_sub_group;
```

Comment:
I think we should combine these into one hint.  We can call it `full_group` for now.

We already know whether we're using a group or a sub-group from the first argument.  There are also going to be other group types in the future, and it would be convenient to be able to re-use the same property across all of those group types as well.

Reply:
I'm not sure `full_group` is a good name. `parallel_for` requires that global size is a multiple of a work group size, so WG is always "full" in some sense. It might still be that the last SG of each WG isn't full.

Reply:
Right, but that just means that `full_group` doesn't do anything when the algorithm is invoked with a `sycl::group`, but improves performance when the algorithm is invoked with a `sycl::sub_group`.  The "group" in "full group" is intended to mean "anything satisfying the group concept" and not "`sycl::group`".

I want to push for a renaming of `sycl::group` to `sycl::work_group` to avoid this sort of confusion.

Reply:
Then we need another property for the WG case, I think.

Reply:
Why?  Maybe I'm misunderstanding what you're expecting to happen here.  I thought your earlier message was saying that work-groups were always "full"?

Reply:
They are, but the block-read is SG-level intrinsic. We need SGs to be "full" to use them.

Reply:
I think we're talking past each other.

Consider the below:

```c++
auto sg = it.get_sub_group();
group_load(sg, input, output_span, properties{contiguous_memory, full_group});
```

The interpretation here is that `sg` represents a full sub-group.

Reply:
If the user writes

```
  auto g = it.get_group();
  group_load(g, input, output_span, properties{contiguous_memory, full_group});
```

we still can't use block loads without runtime checks. WG is full, but its last SG might not be. Also, `full_group` is meaningless for WG because it always full per SYCL spec `parallel_for`/`nd_range` restrictions.

Reply:
We couldn't use block loads in this case anyway, because the stride in this case is the size of the work-group and not the sub-group.  You're right that we could generate block loads in some cases (like a work-group + blocked data placement) but I think we'd still need a runtime check because different devices carve up work-groups into sub-groups differently.

`full_group` is currently meaningless for work-groups, but I think it makes more sense to have a generic `full_group` than to make the property sub-group specific.  That allows a developer to write generic code that uses `full_group` and have it work in both cases, and it may be meaningful for future group types.